tests/virtual_machines_test.go

```diff
@@ -41,44 +44,409 @@ var _ = Describe("Virtual Machines", func() {
+					Expect(err).ToNot(HaveOccurred())
 
-			Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).ToNot(BeEmpty())
-			_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
-			Expect(err).ToNot(HaveOccurred())
+					vmOverlap := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{ovsInterface}, []kubevirtv1.Network{ovsNetwork})
+					// Allocated the same MAC address that was registered to the first vm
+					Expect(vmOverlap.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).To(BeEmpty())
+					vmOverlap.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress = vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+					Expect(vmOverlap.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).ToNot(BeEmpty())
+
+					err = testClient.VirtClient.Create(context.TODO(), vmOverlap)
+					Expect(err).To(HaveOccurred())
+				})
+			})
+			//2167
+			Context("When the MAC address is out of range", func() {
+				It("should reject a vm creation with an already allocated MAC address", func() {
+
+					outOfRangeOvsInterface := kubevirtv1.Interface{
+						Name: "ovs",
+						InterfaceBindingMethod: kubevirtv1.InterfaceBindingMethod{
+							Bridge: &kubevirtv1.InterfaceBridge{},
+						},
+						MacAddress: "10:00:00:00:00:00",
+					}
+
+					vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{outOfRangeOvsInterface}, []kubevirtv1.Network{ovsNetwork})
+					Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).ToNot(BeEmpty())
+
+					Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).ToNot(BeEmpty())
+					_, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+					Expect(err).ToNot(HaveOccurred())
+
+					err = testClient.VirtClient.Create(context.TODO(), vm)
+					Expect(err).ToNot(HaveOccurred())
+
+					vmOverlap := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{outOfRangeOvsInterface}, []kubevirtv1.Network{ovsNetwork})
+					// Allocated the same mac address that was registered to the first vm
+					vmOverlap.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress = vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+
+					err = testClient.VirtClient.Create(context.TODO(), vm)
+					Expect(err).To(HaveOccurred())
+				})
+
+			})
+		})
+		//2199
+		Context("when the client tries to assign the same MAC address for two different interfaces in a single VM.", func() {
+			Context("When the MAC address is within range", func() {
+				It("should	reject a VM creation with two interfaces that share the same MAC address", func() {
+
+					ovsInterface1 := kubevirtv1.Interface{
+						Name: "ovs",
+						InterfaceBindingMethod: kubevirtv1.InterfaceBindingMethod{
+							Bridge: &kubevirtv1.InterfaceBridge{},
+						},
+						MacAddress: "02:00:ff:ff:ff:ff",
+					}
+
+					ovsInterface2 := kubevirtv1.Interface{
+						Name: "ovs",
```

Comment:
every interface and network in kubevirt must be unique  please change the name to `ovs1` and `ovs2`

related error:
```
      Expected error:
          <*errors.StatusError | 0xc00012f3b0>: {
              ErrStatus: {
                  TypeMeta: {Kind: "", APIVersion: ""},
                  ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
                  Status: "Failure",
                  Message: "admission webhook \"virtualmachine-validator.kubevirt.io\" denied the request: Only one interface can be connected to one specific network, Only one interface can be connected to one specific network, Only one interface can be connected to one specific network",
                  Reason: "Invalid",
                  Details: {
                      Name: "",
                      Group: "",
                      Kind: "",
                      UID: "",
                      Causes: [
                          {
                              Type: "FieldValueDuplicate",
                              Message: "Only one interface can be connected to one specific network",
                              Field: "spec.template.spec.domain.devices.interfaces[1].name",
                          },
                          {
                              Type: "FieldValueDuplicate",
                              Message: "Only one interface can be connected to one specific network",
                              Field: "spec.template.spec.domain.devices.interfaces[2].name",
                          },
                          {
                              Type: "FieldValueDuplicate",
                              Message: "Only one interface can be connected to one specific network",
                              Field: "spec.template.spec.domain.devices.interfaces[3].name",
                          },
                      ],
                      RetryAfterSeconds: 0,
                  },
                  Code: 422,
              },
          }
          admission webhook "virtualmachine-validator.kubevirt.io" denied the request: Only one interface can be connected to one specific network, Only one interface can be connected to one specific network, Only one interface can be connected to one specific network
      not to have occurred
```